extension/src/extension-options/work-days-and-hours.ts

```diff
@@ -0,0 +1,99 @@
+import {
+  DayOfWeek,
+  TimeOnly,
+  convertInputTimeToTimeOnly,
+  convertTimeOnlyToInputTime,
+  isTimeBefore,
+  isTimeSame,
+} from '@src/types/dates'
+import { ExtensionOptions } from '@src/types/extension-options'
+import { validator } from './validator'
+
+export const isValidTimes = (): boolean => {
+  const workingHoursStartElement = document.getElementById('working-hours-start') as HTMLInputElement
+  const workingHoursEndElement = document.getElementById('working-hours-end') as HTMLInputElement
+
+  const workStartTime = convertInputTimeToTimeOnly(workingHoursStartElement.value)
+  const workEndTime = convertInputTimeToTimeOnly(workingHoursEndElement.value)
+
+  if (!workStartTime || !workEndTime) {
+    return false
+  }
+
+  if (isTimeSame(workStartTime, workEndTime)) {
+    return false
+  }
+
+  if (isTimeBefore(workEndTime, workStartTime)) {
+    return false
+  }
+
+  return true
+}
+
+const validate = () => {
+  const workingDaysErrorElement = document.getElementById('working-days-error') as HTMLDivElement
+  const workingHoursErrorElement = document.getElementById('working-hours-error') as HTMLDivElement
+
+  const validDays = document.getElementById('working-days').querySelectorAll('button.selected').length > 0
+  validator.setValid('work-days', validDays)
+
+  if (validDays) {
+    workingDaysErrorElement.classList.add('hidden')
+  } else {
+    workingDaysErrorElement.classList.remove('hidden')
+  }
+
+  const validTimes = isValidTimes()
+  validator.setValid('work-times', isValidTimes())
+
+  if (validTimes) {
+    workingHoursErrorElement.classList.add('hidden')
+  } else {
+    workingHoursErrorElement.classList.remove('hidden')
+  }
+}
+
+export const createWorkDaysAndHoursSelectors = (extensionOptions: ExtensionOptions) => {
+  const workingDaysElement = document.getElementById('working-days') as HTMLDivElement
+  const workingDaysButtons = workingDaysElement.querySelectorAll('button') as NodeListOf<HTMLButtonElement>
+  const workingHoursStartElement = document.getElementById('working-hours-start') as HTMLInputElement
+  const workingHoursEndElement = document.getElementById('working-hours-end') as HTMLInputElement
+
+  workingDaysButtons.forEach((button) => {
+    const buttonDayOfWeek = button.getAttribute('data-day-of-week') as DayOfWeek
+
+    if (extensionOptions.workDaysOfWeek.includes(buttonDayOfWeek)) {
+      button.classList.add('selected')
+    }
+
+    button.addEventListener('click', (e) => {
+      e.preventDefault()
+      button.classList.toggle('selected')
+      validate()
+    })
+  })
+
+  workingHoursStartElement.value = convertTimeOnlyToInputTime(extensionOptions.workStartTime)
+  workingHoursEndElement.value = convertTimeOnlyToInputTime(extensionOptions.workEndTime)
+
+  workingHoursStartElement.addEventListener('change', validate)
+  workingHoursEndElement.addEventListener('change', validate)
```

Comment:
Instead of adding event listeners for each element, you should be able to add a change event listener to the entire form. As a bonus, you could then use [FormData](https://developer.mozilla.org/en-US/docs/Web/API/FormData) to get the values of each input, rather than reading each element from the dom.

Reply:
Oh wow that's a good idea. I might look at this in a future PR but let me see what it would look like to do this in a little experimenting locally.

Reply:
Revisiting in the future, thank you!